src/shutdown/umount.c

```diff
@@ -632,23 +636,14 @@ static int umount_with_timeout(MountPoint *m, bool last_try) {
                  * -EIO rather than blocking indefinitely. If the filesysten is "busy", this may allow
                  * processes to die, thus making the filesystem less busy so the unmount might succeed
                  * (rather than return EBUSY). */
-                r = RET_NERRNO(umount2(m->path, MNT_FORCE));
+                r = RET_NERRNO(umount2(m->path,
+                                       UMOUNT_NOFOLLOW | /* Don't follow symlinks: this should never happen unless our mount list was wrong */
+                                       (m->umount_lazily ? MNT_DETACH : MNT_FORCE)));
```

Comment:
Maybe make a `umount_flags` field instead, like `remount_flags`?

Reply:
I thought about this, but I think we probably should start mangling the flags in the loop here sooner or later, i.e. first try to unmount without MNT_FORCE, and then add this in in a later loop. Thus pre-calculating the flags early doesn't really make too much sense, if you have to mangle them later anyway.

But I didn't want to add the conditional MNT_FORCE stuff into the current PR already, and do that some other time, as it is unrelated.